app-misc/lcd4linux/lcd4linux-0.11.0_pre20170527.ebuild

```diff
@@ -0,0 +1,163 @@
+	proc_stat python qnaplog raspi sample statfs uname uptime
+	w1retap wireless )
+
+# Add supported drivers from 'IUSE_LCD_DEVICES' to 'IUSE' and 'REQUIRED_USE'
+REQUIRED_USE+=" || ( "
+for lcd_device in ${IUSE_LCD_DEVICES[@]}; do
+	IUSE="${IUSE} lcd_devices_${lcd_device%%:*}"
+	REQUIRED_USE+="lcd_devices_${lcd_device%%:*} "
+done
+REQUIRED_USE+=" )"
+
+# Add supported plugins from 'IUSE_LCD4LINUX_PLUGINS' to 'IUSE' and 'REQUIRED_USE'
+IUSE="${IUSE} $(printf '%s ' ${IUSE_LCD4LINUX_PLUGINS[@]})"
+REQUIRED_USE="${REQUIRED_USE} || ( $(printf '%s ' ${IUSE_LCD4LINUX_PLUGINS[@]}) )"
+
+# Define dependencies for all drivers in 'IUSE_LCD_DEVICES'
+DEPEND_LCD_DEVICES="lcd_devices_astusb? ( virtual/libusb:0= )
+	lcd_devices_bwct? ( virtual/libusb:0= )
+	lcd_devices_curses? ( sys-libs/ncurses:0= )
+	lcd_devices_dpf? ( virtual/libusb:0= )
+	lcd_devices_g15? ( virtual/libusb:0= )
+	lcd_devices_glcd2usb? ( virtual/libusb:0= )
+	lcd_devices_irlcd? ( virtual/libusb:0= )
+	lcd_devices_lcd2usb? ( virtual/libusb:0= )
+	lcd_devices_ledmatrix? ( virtual/libusb:0= )
+	lcd_devices_luise? ( dev-libs/luise-bin
+		virtual/libusb:0= )
+	lcd_devices_matrixorbitalgx? ( virtual/libusb:0= )
+	lcd_devices_mdm166a? ( virtual/libusb:0= )
+	lcd_devices_picolcd? ( virtual/libusb:0= )
+	lcd_devices_picolcdgraphic? ( virtual/libusb:0= )
+	lcd_devices_png? ( media-libs/gd[png]
+		media-libs/libpng:0= )
+	lcd_devices_ppm? ( media-libs/gd )
+	lcd_devices_samsungspf? ( virtual/libusb:0= )
+	lcd_devices_serdisplib? ( dev-libs/serdisplib )
+	lcd_devices_shuttlevfd? ( virtual/libusb:0= )
+	lcd_devices_trefon? ( virtual/libusb:0= )
+	lcd_devices_ula200? ( dev-embedded/libftdi:1=
+		virtual/libusb:0= )
+	lcd_devices_usbhub? ( virtual/libusb:0= )
+	lcd_devices_usblcd? ( virtual/libusb:0= )
+	lcd_devices_vnc? ( net-libs/libvncserver )
+	lcd_devices_x11? ( x11-libs/libX11 )"
+
+# Define dependencies for all plugins in 'IUSE_LCD4LINUX_PLUGINS'
+DEPEND_LCD4LINUX_PLUGINS="asterisk? ( net-misc/asterisk )
+	dbus? ( sys-apps/dbus )
+	gps? ( dev-libs/nmeap )
+	hddtemp? ( app-admin/hddtemp )
+	iconv? ( virtual/libiconv )
+	mpd? ( media-libs/libmpd )
+	mpris_dbus? ( sys-apps/dbus )
+	mysql? ( || ( dev-db/mariadb-connector-c
+		dev-db/mysql-connector-c ) )
+	python? ( ${PYTHON_DEPS} )
+	wireless? ( || ( net-wireless/iw
+		net-wireless/wireless-tools ) )"
+
+RDEPEND="dmalloc? ( dev-libs/dmalloc )
+	${DEPEND_LCD_DEVICES}
+	${DEPEND_LCD4LINUX_PLUGINS}"
+
+DEPEND="${RDEPEND}"
+
+pkg_setup() {
+	if use python; then
+		python-single-r1_pkg_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	eautoreconf
+}
+
+src_configure() {
+	# Enable all users enabled lcd devices
+	local myeconfargs_lcd_devices
+	for lcd_device in "${IUSE_LCD_DEVICES[@]}"; do
+		if use "lcd_devices_${lcd_device%%:*}"; then
+			myeconfargs_lcd_devices+=",${lcd_device#*:}"
+		fi
+	done
+
+	# Enable all users enabled lcd4linux plugins
+	local myeconfargs_lcd4linux_plugins
+	for lcd4linux_plugin in "${!IUSE_LCD4LINUX_PLUGINS[@]}"; do
```

Comment:
You don't have to do the `!` indirection here?

Reply:
Hm, more, that won't work completly. Rewrote it. I can see now the plugins passed correctly.